test/katex-spec.js

```diff
@@ -2378,3 +2384,54 @@ describe("The maxSize setting", function() {
         expect(built.style.borderTopWidth).toEqual("0em");
     });
 });
+
+describe("Tree attributes propagation", function() {
+    it("should put tree attributes to DOM nodes", function() {
+        const expr = "\\frac 1 2";
+        const tree = getParsed(expr);
+
+        tree[0].attributes = {
+            "katex-frac-id": "frac",
+        };
+
+        tree[0].value.numer.attributes = {
+            "katex-numer-id": "numer",
+        };
+
+        tree[0].value.denom.attributes = {
+            "katex-denom-id": "denom",
+        };
+
+        let counter = 1;
+        const putId = function(treeNode) {
+            if (!treeNode.attributes) {
+                treeNode.attributes = {};
```

Comment:
I would probably do:
```
let counter = 1;
const putId = (treeNode, htmlNode) => {
    const id = count++;
    treeNode.id = id;
    if (htmlNode instanceof BaseNode) {
         htmlNode.properties['katex-id'] = id;
    }
}
```
This assumes that htmlNode is a HAST node, see https://github.com/syntax-tree/hast and https://github.com/Khan/KaTeX/issues/800.  This allows us to avoid having to create `defaultPostProcessor` in buildHTML.js which splits up the logic of what we're trying to do across multiple files.